test agreement phase of market interaction

tworec opened the pull request:
resolves #431

tworec commented on core/market/decentralized/src/testing/proposal_util.rs:
```diff
@@ -9,54 +9,56 @@ use crate::testing::OwnerType;
         .await?;
 
     // Provider receives Proposal
-    let events = market2
+    let prov_events = prov_mkt
         .provider_engine
         .query_events(&offer_id, 1.2, Some(5))
         .await?;
-    let proposal1_prov = provider::expect_proposal(events)?;
-    let proposal1_prov_id = proposal1_req_id.clone().translate(OwnerType::Provider);
+    let demand_proposal2 = provider::expect_proposal(prov_events)?;
+    let demand_proposal2_id = demand_proposal1_id.clone().translate(OwnerType::Provider);
 
     // Provider counters proposal.
-    let proposal2_prov = proposal1_prov.counter_offer(sample_offer())?;
-    let _proposal2_id = market2
+    let offer_proposal2 = demand_proposal2.counter_offer(sample_offer())?;
+    let _offer_proposal_id = prov_mkt
         .provider_engine
-        .counter_proposal(&offer_id, &proposal1_prov_id, &proposal2_prov)
+        .counter_proposal(&offer_id, &demand_proposal2_id, &offer_proposal2)
         .await?;
 
     // Requestor receives proposal.
-    let events = market1
+    let req_events = req_mkt
         .requestor_engine
         .query_events(&demand_id, 1.2, Some(5))
         .await?;
-    let proposal2_req = requestor::expect_proposal(events)?;
-    Ok(ProposalId::from_str(&proposal2_req.proposal_id.unwrap())?)
+    let offer_proposal3 = requestor::expect_proposal(req_events)?;
+    Ok(ProposalId::from_str(&offer_proposal3.proposal_id.unwrap())?)
 }
```
I was trying to make it more readable, because `proposal1_prov_id` resembles rather `provider_id` than provider side of proposal.
I will rename again then as `demand_proposal1` to `req_demand_proposal1`
and `demand_proposal2` to `prov_demand_proposal`, ok?

nieznanysprawiciel replied:
It could be.
It's not blocking problem

tworec commented on core/market/decentralized/tests/test_agreement.rs:
```diff
@@ -43,3 +44,330 @@ async fn test_gsb_get_agreement() -> Result<(), anyhow::Error> {
+        .approve_agreement(prov_id.clone(), &agreement_id, 0.1)
+        .await?;
+
+    // TODO: is it really ok to allow such situation?
+    // shouldn't we return an error here?
+    // assert_eq!(
+    //     result.unwrap_err().to_string(),
+    //     AgreementError::ProtocolApprove(..).to_string()
+    // );
+
```
> Not only allowed, but we must do it this way because otherwise, we will have races. Provider shouldn't depend on Requestor behavior. And if we don't allow to approve, before wait call on requestor side, Provider must sleep until Requestor will call it??? It would be horrible from Provider implementer perspective.

I'm not so sure we must. Both provider and requestor specifies `timeout`. One of timeout reasons is that the other side is not called approval/rejection. But Requestor's side is not symmetrical to Provider's here, since Requestor need to: 
1. create agreement
1. **confirm** it 
1. wait for approval

Confirmation is blocking approval on Provider side. This is enough. I can sleep soundly ;)

> I think you could check in this test `wait_for_approval` after `approve_agreement` scenario, which I by the way implemented (remember this loop in implementation? It was for this case)

ok!

nieznanysprawiciel replied:
But still I don't think Provider should get error, if it called approve earlier.

tworec replied:
done

tworec commented:
> It is good set of tests ;)
> We have probably 3 cases left:
> 
> * double `create_agreement` (It is allowed in general, but probably after rejection or expiration??, but we don't know even how we should handle this case)

added as ignored case

> * double `wait_for_approval` (at the same time)

added 

> * `wait_for_approval` after first `wait_for_approval` succeeded

added

jiivan commented on core/market/decentralized/tests/test_agreement.rs:
```diff
@@ -1,45 +1,615 @@
+//#[cfg_attr(not(feature = "market-test-suite"), ignore)]
+#[ignore]
+#[actix_rt::test]
+async fn second_creation_should_fail() -> Result<()> {
```
Is it possible to have a name conflict with test from other teams?

tworec replied:
nope, tests need to be unique just within single file or even module
